backtrace-library/src/main/java/backtraceio/library/logger/BacktraceLogger.java

```diff
@@ -1,25 +1,25 @@
 package backtraceio.library.logger;
 
-import android.util.Log;
-
 /**
- * Backtrace Logger class for logging messages from inside library
+ * A global logging utility for the Backtrace library.
+ * <p>
+ * {@code BacktraceLogger} acts as a wrapper around a {@link Logger} implementation,
+ * providing a centralized logging mechanism throughout the library. By default, it uses
+ * the {@link BacktraceLogLogger}, which relies on {@code android.util.Log} for logging.
+ * However, the logger can be replaced with a custom implementation by using the
+ * {@link #setLogger(Logger)} method.
+ * </p>
+ * <p>
+ * This allows for flexibility in logging strategies, enabling developers to integrate
+ * their preferred logging framework or customize log handling as needed.
+ * </p>
  */
 public class BacktraceLogger {
 
-    private static final String BASE_TAG = "BacktraceLogger: ";
-    /**
-     * Level from which all information is logged
-     */
-    private static LogLevel logLevel = LogLevel.OFF;
+    private static Logger logger = new BacktraceLogLogger();
 
-    /**
-     * set logging level from which all messages should be logged to the console
-     *
-     * @param level login level
-     */
-    public static void setLevel(LogLevel level) {
```

Comment:
Unfortunately we should keep it. Otherwise this will be a breaking change. I know some customers are using it

Reply:
Good point, added with deprecated annotation